iceoryx_hoofs/source/cxx/axivion.cpp

```diff
@@ -0,0 +1,68 @@
+#include "iceoryx_hoofs/internal/cxx/adaptive_wait.hpp"
```

Comment:
We could also introduce a header called `cxx.hpp` where all public/private headers are included. It would look somehow cleaner. An `axivion.cpp` file looks like we have to perform hacks to support this tool.

Furthermore, for me it is essential that when we have to create such a header collection file this has to be done by script. I will not accept such a manually generated file since it is extremely hard to review (if every header was actually included) and if a new header is added this file will be overlooked and not be adjusted.

I would recommend a simple bash script which iterates over all `iceoryx_hoofs` components and creates a file corresponding to the component name like `cxx` -> `cxx.hpp` or `posix` -> `posix.hpp`. Whenever a new file is added the script has to rerun and adjust those files.
This bash script should also support some type of CI check where it verifies that the header are up to date and fails when they are not.